### PR TITLE
feat: add ssl

### PR DESCRIPTION
**Pull request by laurasgkadri98**
- [x] Add secured ``https`` connection as an option.
- [x] Modify how service is started: give priority to cli and for empty values use config file. As last resort use default values.

### CODE REVIEW

File: src/allie/flowkit/__main__.py [hunk 1]
```diff
@@ -23,20 +23,54 @@
...
+
+def parse_cli_args():
+    """Parse the command line arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--host", type=str, help="The host to run the service on. By default 0.0.0.0")
+    parser.add_argument("--port", type=int, help="The port to run the service on. By default 50052")
+    parser.add_argument("--workers", type=int, help="The number of workers to use. By default 4")
+    parser.add_argument("--use-ssl", type=bool, help="Enable SSL for the service. By default False")
+    parser.add_argument("--ssl-keyfile", type=str, help="The SSL key file path")
+    parser.add_argument("--ssl-certfile", type=str, help="The SSL certificate file path")
```
Comment (RobPasMue) [inline src/allie/flowkit/__main__.py:42]: required=False?

File: src/allie/flowkit/config/_config.py [hunk 2]
```diff
@@ -59,8 +59,16 @@ def __init__(self):
...
         config_path = os.getenv("ALLIE_CONFIG_PATH", "config.yaml")
         self._yaml = self._load_config(config_path)
+
+        # Define the configuration variables to be parsed from the YAML file
         self.flowkit_python_api_key = self._yaml.get("FLOWKIT_PYTHON_API_KEY")
+        self.flowkit_python_endpoint = self._yaml.get("FLOWKIT_PYTHON_ENDPOINT", "http://localhost:50052")
+        self.flowkit_python_workers = self._yaml.get("FLOWKIT_PYTHON_WORKERS", 4)
+        self.use_ssl = self._yaml.get("USE_SSL", False)
+        self.ssl_cert_public_key_file = self._yaml.get("SSL_CERT_PUBLIC_KEY_FILE")
+        self.ssl_cert_private_key_file = self._yaml.get("SSL_CERT_PRIVATE_KEY_FILE")
```
Comment (RobPasMue) [inline src/allie/flowkit/config/_config.py:70]: Aha.. you are defining the defaults here now, correct?

**Review by RobPasMue — APPROVED** [review]
Add the required=False arg, this makes it clear that the args are optional. Apart from that all good =)

File: src/allie/flowkit/config/_config.py (hunk 2 above)
Comment (laurasgkadri98) [reply src/allie/flowkit/config/_config.py:70]: Yes, thats right

**Review by RobPasMue — APPROVED** [review]

**Pull request merged**